### PR TITLE
V14.0 - logistics bunch of bugfixes 

### PR DESCRIPTION
**Pull request by ryv-odoo**
[FIX] stock: fix picking_type_id of stock move.

The picking_type_id of a stock move can differ of the related picking:
- Create and save a draft transfer with a stock move
- Change the picking_type_id and save
- The picking_type of moves won't be the same than the one on the stock
picking.
Fix force the picking_type_id of moves_lines in the
picking `onchange_picking_type` method. Same issue can appear in multi
company situation, fix it too.

[FIX] stock: fix creation of picking type for a other company

In multi company (company A and B),
- Select the company A as main one, and select B too.
- Create a picking type and change the company to B.
- Don't select any warehouse.
- We can't save, because the company ir.sequence is
set to A instead of B.

Fix by apply the company_id of the picking type.

[FIX] mrp: date_planned_start readonly when done

change the unexciting `close` state in `done` in the attrs of
`date_planned_start`

[FIX-IMP] stock: fix and batch push_apply

- Fix the issue of the added test.
- Also batch the `push_apply` + `_action_confirm` by doing confirm
in BFS instead of DFS. Improve a lot the performance when
the _action_confirm is on batch. WIP

[FIX] mrp: company BoM form

Add to `check_company` on `bom_id` of operation, to avoid company issue
after in MO when we create `mrp.workcenter.productivity.loss`.
Also remove the readonly on `company_id` to get the possibility of
modify it if the company_id of BoM change.

task-2372196

### CODE REVIEW

File: addons/mrp/models/mrp_routing.py [hunk 1]
```diff
@@ -16,13 +16,12 @@ class MrpRoutingWorkcenter(models.Model):
...
     bom_id = fields.Many2one(
-        'mrp.bom', 'Bill of Material',
+        'mrp.bom', 'Bill of Material', check_company=True,
         index=True, ondelete='cascade',
         help="The Bill of Material this operation is linked to")
     company_id = fields.Many2one(
         'res.company', 'Company',
-        readonly=True, store=True,
-        default=lambda self: self.env.company)
+        store=True, default=lambda self: self.env.company)
```
Comment (Whenrow) [inline addons/mrp/models/mrp_routing.py:24]: useless store=True, the field is no more computed

File: addons/stock/models/stock_picking.py [hunk 2]
```diff
@@ -590,11 +590,13 @@ def onchange_picking_type(self):
             else:
                 location_dest_id, supplierloc = self.env['stock.warehouse']._get_partner_locations()
 
-            if self.state == 'draft':
-                self.location_id = location_id
-                self.location_dest_id = location_dest_id
+            self.location_id = location_id
+            self.location_dest_id = location_dest_id
+            (self.move_lines | self.move_ids_without_package).update({
```
Comment (Whenrow) [inline addons/stock/models/stock_picking.py:595]: move_lines contains move_ids_without_package

**Comment by amoyaux** [comment]
@ryv-odoo Just move commit 89d1ede06080287ecd27f69386c42f058a08493d to another PR for master.

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**